internal/cloud/realm/auth.go

```diff
@@ -34,14 +35,14 @@ func (c *client) Authenticate(publicAPIKey, privateAPIKey string) (Session, erro
 		http.MethodPost,
 		authenticatePath,
 		authenticateRequest{publicAPIKey, privateAPIKey},
-		api.RequestOptions{},
+		api.RequestOptions{PreventRefresh: true},
```

Comment:
without this param, there are times we hide a more descriptive invalid session message. I felt that this is the one call where it would be fine to not retry. And that way we can return "failed to authenticate with MongoDB Cloud API: You are not authorized for this resource." instead of a generic "invalid session" message if the login credentials are invalid